java/maven/src/org/netbeans/modules/maven/spi/actions/AbstractMavenActionsProvider.java

```diff
@@ -88,6 +88,10 @@ public AbstractMavenActionsProvider() {
             files.add(method.getFile());
         }
 
+        if (files.isEmpty()) {
```

Comment:
Ant projects honor `FileObject` as a selection in the `Lookup`, but Maven projects didn't. After this change they will use `FileObject` (if no `DataObject` is found) as the indentification of main file to execute.

Reply:
Also [fixed in Gradle](034a600c1329).